WalletWasabi/WabiSabi/Client/Batching/PaymentAwareOutputProvider.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using WalletWasabi.Extensions;
+using WalletWasabi.Logging;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+
+namespace WalletWasabi.WabiSabi.Client.Batching;
+
+// Represents an `OutputProvider` that has a reference to the `BatchedPayments` instance created by a `Wallet`.
+// This class is then aware of the existence of payments for a `Wallet`, what allows it to provide outputs for
+// coinjoin round which include those outputs that are payments.
+public class PaymentAwareOutputProvider : OutputProvider
+{
+	public PaymentAwareOutputProvider(IDestinationProvider destinationProvider, PaymentBatch batchedPayments)
+		: base(destinationProvider)
+	{
+		BatchedPayments = batchedPayments;
+	}
+
+	private PaymentBatch BatchedPayments { get; }
```

Comment:
The type and the instance should in most cases be named differently because it is easier to understand `Person mother`, `Counter duplicates` than `Person person`,  `Counter counter`.